aggregate/aggregate_count.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aggregate
+
+import (
+	"github.com/hazelcast/hazelcast-go-client/internal"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+// Count returns the number of values which includes the given attribute.
+// Note that this function may not work as expected for Hazelcast versions prior to 5.0.
+func Count(attr string) *aggCount {
+	return &aggCount{attrPath: attr}
+}
+
+// CountAll returns the number of all values.
+func CountAll() *aggCount {
```

Comment:
So, `Count("")` is equal to `CountAll()`, right? I would personally be okay with just `Count(attr)` and document that if you pass `""`, we will count all values.

But I am okay with this approach as well. I just wasn't 100% sure about having a different method just for one variant of the input.

Reply:
Yeah, `CountAll()` is easier to discover than `Count("")`.